src/pip/_internal/vcs/bazaar.py

```diff
@@ -56,14 +53,15 @@ def fetch_new(self, dest, url, rev_options):
             rev_display,
             display_path(dest),
         )
-        cmd_args = ['branch', '-q'] + rev_options.to_args() + [url, dest]
+        cmd_args = (['checkout', '--lightweight', '-q'] +
+                    rev_options.to_args() + [url, dest])
         self.run_command(cmd_args)
 
     def switch(self, dest, url, rev_options):
         self.run_command(['switch', url], cwd=dest)
 
-    def update(self, dest, url, rev_options):
-        cmd_args = ['pull', '-q'] + rev_options.to_args()
+    def update(self, dest, rev_options):
```

Comment:
You need to keep the signature of `update()` the same because that's how it's invoked in the base class's `obtain()` method.